### ISSUE TITLE
label value string with commas - Couldn't parse LABELS

### ISSUE THREAD

**Issue by earthgecko** [RedisTimeSeries/RedisTimeSeries#1478]
Hi

Using redis-stack-server (redis-stack-server-6.2.6-v7) on Ubuntu 22.04 installed from a binary download which in `INFO` reports:

```
# Server
redis_version:6.2.12
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:c4787fdd57f68bb3
redis_mode:standalone
os:Linux 5.15.0-67-generic x86_64
arch_bits:64
monotonic_clock:POSIX clock_gettime
multiplexing_api:epoll
atomicvar_api:c11-builtin
gcc_version:11.3.0
...
...
# Modules
module:name=timeseries,ver=10810,api=1,filters=0,usedby=[],using=[],options=[handle-io-errors]
```

I have run into an unexpected issue related to a label value string with commas in it resulting in `TSDB: Couldn't parse LABELS`

Using py-redis (redis==4.6.0) and issueing: 
```
redis_conn.ts().add(redis_ts_key, timestamp_ms, value, retention_msecs=retention_msecs, labels=labels, chunk_size=128, duplicate_policy='first')
```

Where labels are:
`{'labels': {'__name__': 'prometheus_build_info', 'branch': 'HEAD', 'goarch': 'amd64', 'goos': 'linux', 'goversion': 'go1.20.5', 'instance': 'localhost:9090', 'job': 'prometheus', 'revision': '8ef767e396bf8445f009f945b0162fd71827f445', 'tags': 'netgo,builtinassets,stringlabels', 'version': '2.45.0'}}`

Results in `ResponseError("TSDB: Couldn't parse LABELS")`

However if I do a replace on the string and replace the commas with spaces on the tags label value, changing `'netgo,builtinassets,stringlabels'` to `'netgo builtinassets stringlabels'` the `ts().add` works.

Is this the expected behaviour in RedisTimeSeries?  I have not not found any clear or definitive documentation on what formats/chars are allowed in the label values.

**Comment by LiorKogan** [comment]
Hello @earthgecko 

We can't allow commas as part of a label name.

Consider the command `MRANGE ... FILTER label=(value1,value2) ...`

How can we tell if theses are two values (`value1` and `value2`) or a single label name (`value1,value2`)?

**Comment by OfirMos** [comment]
Hi @earthgecko,
You should use parentheses around multiple label values, i.e:
{‘labels’: {‘__name__‘: ‘prometheus_build_info’, ‘branch’: ‘HEAD’, ‘goarch’: ‘amd64’, ‘goos’: ‘linux’, ‘goversion’: ‘go1.20.5’, ‘instance’: ‘localhost:9090’, ‘job’: ‘prometheus’, ‘revision’: ‘8ef767e396bf8445f009f945b0162fd71827f445’, ‘tags’: ‘(netgo,builtinassets,stringlabels)’, ‘version’: ‘2.45.0’}}

**Comment by LiorKogan** [comment]
Hello @earthgecko 

We can't allow commas as part of a label name.

Consider the command `MRANGE ... FILTER label=(value1,value2) ...`

How can we tell if theses are two values (`value1` and `value2`) or a single value (`value1,value2`)?

**Comment by earthgecko** [comment]
Hi @LiorKogan I am a bit confused now.... Yesterday your comment was removed regarding commas and then a comment was made by @OfirMos saying:
> You should use parentheses around multiple label values

Now that @OfirMos comment has been removed and @LiorKogan you have added this comment saying wrapping in `()` nor `,` are allowed.

I am not sure why comments on this issue keep on being deleted, it makes it hard to follow, but I can see the dilemma.

I was wondering yesterday if this issue was maybe handled by https://github.com/RedisTimeSeries/prometheus-redistimeseries-adapter but looking at the code it does not seem to be handled.

So basically replace `,` in label values is the only viable solution it seems.

Thanks for the clarification and correction.

**Comment by LiorKogan** [comment]
Sorry for the confusion. Originally @OfirMos thought I was wrong, but it turned out I was correct. That's why we removed and readded the comments.

Yes, you'll have to replace the `,` in the label values.
